skimage/morphology/_flood_fill.py

```diff
@@ -230,22 +230,29 @@ def flood(image, seed_point, *, footprint=None, connectivity=None,
     seed_value = image[seed_point]
     seed_point = tuple(np.asarray(seed_point) % image.shape)
 
-    footprint = _resolve_neighborhood(footprint, connectivity, image.ndim)
+    footprint = _resolve_neighborhood(
+        footprint, connectivity, image.ndim, enforce_adjacency=False)
+    center = np.array(footprint.shape) // 2
+
+    # Compute pad count as the maximum offset to neighbors across all axes.
+    neighbor_indices = np.array(np.nonzero(footprint)).T
+    neighbor_coordinate_offsets = np.ravel(neighbor_indices - center)
+    pad_count = np.max(np.abs(neighbor_coordinate_offsets))
```

Comment:
Having padding that is > 1 is a change for any existing code that uses an adjacent footprint, but with size > 3 along any axis. I am wondering if we should autodetect adjacency in the footprint and use a padding of 1 whenever it is adjacent?

Reply:
Just to check that I'm correctly understanding your comment: you're referencing cases where the footprint looks like: [[0, 1, 1, 1, 0], [0, 1, 1, 1, 0], [0, 1, 1, 1, 0]]. Where only adjacent points are marked as neighbors, but the footprint still has size > 3 along an axis. 
In this case I think the padding should still equal 1, because neighbor_indices only includes indices with nonzero values in the footprint.

Reply:
I thought about it again, and I believe that the current implementation already autodetects adjacency in a sense. If only adjacent entries in footprint are equal to 1 then pad count should always end up equaling 1, as only nonzero entries of footprint are considered in the pad_count calculation.